rest_framework/fields.py

```diff
@@ -720,6 +720,12 @@ class BooleanField(Field):
     }
     NULL_VALUES = {'null', 'Null', 'NULL', '', None}
 
+    def __init__(self, **kwargs):
+        if 'allow_null' in kwargs:
```

Comment:
what if `allow_null=False`?

Reply:
> As I understand, right now if you want to migrate and preserve the initial value being None, you need NullBooleanField(allow_null=True, initial=None) ? I don't think we need to change BooleanField for this.

That's the problem, NullBooleanField (NBF) is deprecated but the current implementation of BooleanField does not fully support null values. NBF will be removed from Django eventually and DRF won't be aligned.

> I don't quite follow why the `default_empty_html` change is needed - it doesn't look like `NullBooleanField` ever had that:

Because otherwise it will be treated as False instead of None

Reply:
That would be the fix, the template for BooleanField did not allow to set it to None and treated it as False. This was spotted here: https://github.com/math-a3k/covid-ht/blob/master/data/serializers.py#L14

The current implementation needs those in order to behave as expected - otherwise the current implementation will treat Nones as Falses. Without this patch, this test will fail: https://github.com/encode/django-rest-framework/pull/8614/files#diff-a828467ae280cb3ae472f4c3fa7fafe22f9e672829114f1ecccc14435fa205f8R394 (initial=None is not necessary, just allow_null=True)

Reply:
we have some pointer here in this comment https://github.com/encode/django-rest-framework/issues/7582#issuecomment-708905051